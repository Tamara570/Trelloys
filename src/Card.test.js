import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Card from './card';


describe(`Card component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders markup as expected without props', () => {
        const wrapper = shallow(<Card />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

    it('renders Card when supplied with an items prop', () => {
        const testItem = { id: 'a', title: 'First card', content: 'lorem ipsum' }
        const wrapper = shallow(<Card item={testItem} />)
        expect(toJson(wrapper)).toMatchSnapshot()
      })
})


