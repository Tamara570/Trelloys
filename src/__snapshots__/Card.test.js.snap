// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Card when supplied with an items prop 1`] = `
<div
  className="Card"
>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;

exports[`Card component renders markup as expected without props 1`] = `
<div
  className="Card"
>
  <button
    onClick={[Function]}
    type="button"
  >
    delete
  </button>
  <h3 />
  <p />
</div>
`;
